soc/nordic/CMakeLists.txt

```
@@ -17,11 +17,14 @@ zephyr_library_sources(
 # headers for different SoCs.
 set(dt_binding_includes ${DTS_INCLUDE_FILES})
 list(FILTER dt_binding_includes INCLUDE REGEX "/dt-bindings/.*\.h$")
-list(TRANSFORM dt_binding_includes PREPEND "-include;")
+
+get_target_property(include_flag compiler include_file)
+set(include_flag ${include_flag}$<SEMICOLON>)
```

please switch to using a genex. genex'es are evaluated at generator time whereas `get_target_property()` is evaluated at configure time. Thus genex is a bit safer, and should be used when possible.

```suggestion
set(include_flag $<TARGET_PROPERTY:compiler,include_file>$<SEMICOLON>)
```